runner/src/rpc/tool.rs

```diff
@@ -96,3 +107,18 @@ pub(crate) fn credentials_filepath(account_id: AccountId) -> Result<PathBuf, Str
     path.push(format!("{}.json", account_id));
     Ok(path)
 }
+
+/// Convert `StateItem`s over to a Map<data_key, value_bytes> representation.
+/// Assumes key and value are base64 encoded, so this also decode them.
+pub(crate) fn into_state_map(state_items: Vec<StateItem>) -> HashMap<String, Vec<u8>> {
+    let decode = |s: StateItem| {
+        (
+            str::from_utf8(&base64::decode(s.key.clone()).unwrap())
+                .unwrap_or_else(|_| &s.key)
+                .to_owned(),
+            base64::decode(s.value).unwrap(),
+        )
+    };
+
+    state_items.into_iter().map(decode).collect()
```

Comment:
You shouldn't need owned values for anything in the decode function, perhaps this would be more generally usable if the function just took a slice of elements and this just iterated through them?

Reply:
That's a good point, unnecessary ownership is equally bad here. Fixed